### DISCUSSION TITLE
changing the prompt in dmenu mode on the fly

### DISCUSSION THREAD

**Discussion by ghost** [davatorium/rofi#1649] [General]
I noticed I can't change rofi's prompt on the fly from a script using `echo -e "\0prompt\x1fsomething"`
when running rofi with `-dmenu` and `-async-pre-read 0` flag.

I want to change the prompt on the fly because my script will fetch some data from an online api and I want to change prompt after all options have been loaded. ( change prompt from "loading" to "done" for example) similar to how fzf shows a loading character before the input. so that I can know when all the options have been fetched

**Comment by DaveDavenport** [comment]
Add the overlay widget to your theme?
```bash
while true ; do echo "a"; sleep 1; done| ./rofi -dmenu -theme-str 'inputbar { children: [prompt,entry,overlay,case-indicator]; }'  -no-config -async-pre-read 0
```

**Reply by ghost** [reply]
that works thanks.
is it possible to change the overlay message?

**Reply by DaveDavenport** [reply]
no

**Reply by ghost** [reply]
is there any customizability for it? changing it's font, or color for example?

**Reply by ghost** [reply]
i tried `background-color`, `border`, `font` and `color`. and they all worked.
thanks for your quick response!
